packages/backend-api/src/api/services/updateNotifications.ts

```diff
@@ -122,6 +122,28 @@ async function getSystemData() {
   } as IMessage;
 }
 
+// Convert IDs to strings, and assignedModerators to arrays of strings.
+function serialiseObject(o: any, fields: Array<string>): {[key: string]: any} {
```

Comment:
Prefer `{}` over `any` unless you really need any. And in this case you don't, and you require that `o` has at least the `id` field. Also the output here will also have the `id` field, and optionally has as `assignedModerators`, and will have it as a string. It is better to have functions more directly specified, and then have callers of the function do type coercions than it is to abuse types in leaf functions. If you make each leaf function correctly typed, then you can push the correct types up the code stack bit by bit. But you have to start somewhere. :)

Reply:
I updated the interface.  You are very much correct that we can give a much better type to o - its a sequelize object.

WRT the return type, I don't think we need to care that it has id and assignedModerators attributes.  What it is returning is a JSONable blob.   Other versions of this function may return blobs with a completely different set of attributes.  I've tightened up the type a bit to indicate that that is the case...